db.go

```diff
@@ -213,6 +215,93 @@ func (db *DB) Get(ro *ReadOptions, key []byte) ([]byte, error) {
 	return C.GoBytes(unsafe.Pointer(value), C.int(vallen)), nil
 }
 
+// GetMany returns the values associated with keys from the database.
+//
+// GetMany fills in the entries of values consistent with Get():
+// - If keys[i] exists in the database with non-zero value, values[i] holds its non-zero len []byte slice.
+// - If keys[i] does not exist in the database, values[i] == nil.
+// - If keys[i] does exist, but its value is zero-length, values[i] == []byte{}.
+// - If there's an error looking up keys[i], values[i] == nil.
+//
+// The keys byte slice may be reused safely. Go takes a copy of
+// them before returning.
+func (db *DB) GetMany(ro *ReadOptions, keys [][]byte) ([][]byte, []error) {
+	if db.closed {
+		panic(ErrDBClosed)
+	}
+	if len(keys) == 0 {
+		return nil, nil
+	}
+
+	var packedKeysBuffer bytes.Buffer
+	cKeyLens := make([]C.size_t, len(keys))
+	for i := range keys {
+		packedKeysBuffer.Write(keys[i])
+		cKeyLens[i] = C.size_t(len(keys[i]))
+	}
+	packedKeysBytes := packedKeysBuffer.Bytes()
+	if len(packedKeysBytes) == 0 {
+		// If all the keys are empty, abide by the behavior of Get() when given
+		// an empty key and return the same (value, error) pair for all keys[i].
+		// This is the most consistent, though somewhat hilarious, behavior.
```

Comment:
@Viq111 See my proposed `GetMany()` behavior upon getting a list of empty keys :trollface:

Reply:
👍 Mind adding tests and benchmarks to the go side too on this PR ?
(would be great starting to have tests for this lib 🙂 )

Reply:
sounds good. I wrote some here https://github.com/DataDog/dd-go/pull/8658 but it makes sense for them to go into this repo.

Reply:
Ported the tests and benchmarks @Viq111 I think it's reasonably bug free and fast now.